src/global/Constants.h

```diff
@@ -100,13 +100,9 @@ constexpr inline std::string_view MATCHINGWORD_VARIABLE_PREFIX =
 constexpr inline std::string_view LANGUAGE_PREDICATE =
     makeQleverInternalIriConst<"langtag">();
 
-// this predicate is one of the supported identifiers for the SpatialJoin class.
-// It joins the two objects, if their distance is smaller or equal to the
-// maximum distance, which needs to be given in the predicate as well. The
-// syntax for the predicate needs to be like this:
-// <max-distance-in-meters:XXXX>, where XXXX needs to be replaced by an integer
-// number.
+// For backward compatibility: invocation of SpatialJoin via special predicates.
 static const std::string MAX_DIST_IN_METERS = "<max-distance-in-meters:";
+static const std::string NEAREST_NEIGHBORS = "<nearest-neighbors:";
```

Comment:
Would it be feasible, 
to leave the `<nearest-neighbors:>` predicate in for now, and detect (in the queryPlanner), that it was used, and then use `QueryPlanner::addWarning()` to more gracefully ask users to update their query?
This will make it easier to merge this PR, as we don't immediately break example queries on our public instances.

Reply:
I figured that with the bad semantics of `<nearest-neighbors>` it would be reasonable to remove it, but I can readd it if you favor the backward compatibility.